src/queries/database.ts

```diff
@@ -1,209 +1,389 @@
+      allowNull: false,
+    },
+    user_group_id: {
+      type: DataTypes.BIGINT,
+      references: { model: UserGroupModel, key: "iduser_groups" },
+      allowNull: false,
+    },
+  },
+  {
+    tableName: "data_group_memberships",
+    createdAt: false,
+    updatedAt: false,
+  }
+);
+
+const UserGroupMembershipModel = sequelize.define(
+  "UserGroupMembership",
+  {
+    iduser_group_memberships: {
+      type: DataTypes.BIGINT,
+      allowNull: false,
+      primaryKey: true,
+    },
+    user_id: {
+      type: DataTypes.BIGINT,
+      references: { model: UserModel, key: "id" },
+      allowNull: false,
+    },
+    user_group_id: {
+      type: DataTypes.BIGINT,
+      references: { model: UserGroupModel, key: "iduser_groups" },
+      allowNull: false,
+    },
+  },
+  {
+    tableName: "user_group_memberships",
+    createdAt: false,
+    updatedAt: false,
+  }
+);
+
+const ItemTypeModel = sequelize.define(
+  "ItemType",
+  {
+    iditem_types: {
+      type: DataTypes.BIGINT,
+      allowNull: false,
+      primaryKey: true,
+    },
+    name: { type: DataTypes.STRING, allowNull: false },
+    description: { type: DataTypes.STRING, allowNull: false },
+    source: { type: DataTypes.STRING, allowNull: false },
+  },
+  {
+    tableName: "item_types",
+    createdAt: false,
+    updatedAt: false,
+  }
+);
 
-UserModel.hasMany(UserMapModel, { foreignKey: { name: 'user_id' } });
-UserModel.hasMany(PasswordResetTokenModel, { foreignKey: { name: 'user_id' } });
+const MapMembershipModel = sequelize.define(
+  "MapMembership",
+  {
+    idmap_memberships: {
+      type: DataTypes.BIGINT,
+      allowNull: false,
+      autoIncrement: true,
+      primaryKey: true,
+    },
+    map_id: {
+      type: DataTypes.BIGINT,
+      references: { model: MapModel, key: "id" },
+      allowNull: false,
+    },
+    item_type_id: {
+      type: DataTypes.BIGINT,
+      references: { model: ItemTypeModel, key: "iduser_groups" },
+      allowNull: false,
+    },
+    item_id: { type: DataTypes.BIGINT, allowNull: false },
+  },
+  {
+    tableName: "map_memberships",
+    createdAt: false,
+    updatedAt: false,
+  }
+);
 
-MapModel.hasMany(UserMapModel, { foreignKey: { name: 'map_id' } });
-MapModel.hasMany(PendingUserMapModel, { foreignKey: { name: 'map_id' } });
+const PasswordResetTokenModel = sequelize.define(
+  "PasswordResetToken",
+  {
+    idpassword_reset_token: {
+      type: DataTypes.BIGINT,
+      allowNull: false,
+      autoIncrement: true,
+      primaryKey: true,
+    },
+    user_id: {
+      type: DataTypes.BIGINT,
+      allowNull: false,
+      references: { model: UserModel, key: "id" },
+    },
+    token: { type: DataTypes.STRING, allowNull: false },
+    expires: { type: DataTypes.BIGINT, allowNull: false },
+  },
+  {
+    tableName: "password_reset_token",
+    createdAt: false,
+    updatedAt: false,
+  }
+);
+
+const UserFeedbackModel = sequelize.define(
+  "UserFeedback",
+  {
```

Comment:
We need to define the 'id' as the primary auto-incrementing key here (see other model definitions in the file for how this is done)